rasa/core/lock_store.py

```diff
@@ -329,3 +354,155 @@ def _load_from_module_name_in_endpoint_config(
+                f"Omitting provided non-alphanumeric redis key prefix: '{key_prefix}'. "
+                f"Using default '{self.key_prefix}' instead."
+            )
+
+    def get_lock(self, conversation_id: Text) -> Optional[TicketLock]:
+        """Retrieves lock (see parent docstring for more information)."""
+        tickets: Deque[Ticket] = deque()
+
+        pattern = self.key_prefix + conversation_id + ":" + "[0-9]*"
+        redis_keys = self.red.keys(pattern)
+
+        for key in redis_keys:
+            serialised_ticket = self.red.get(key)
+            ticket = Ticket.from_dict(json.loads(serialised_ticket))
+            tickets.appendleft(ticket)
+
+        tickets = deque(sorted(tickets, key=lambda x: x.number))
+
+        return TicketLock(conversation_id, tickets)
+
+    def delete_lock(self, conversation_id: Text) -> None:
+        """Deletes lock for conversation ID."""
+        pattern = self.key_prefix + conversation_id + ":*"
+        redis_keys = self.red.keys(pattern)
+
+        if not redis_keys:
+            logger.debug(
+                f"The lock store does not contain any key-value "
+                f"items for conversation '{conversation_id}'."
+            )
+            return None
+
+        deletion_successful = self.red.delete(*redis_keys)
+        if deletion_successful == 0:
+            self._log_deletion(conversation_id, False)
+        else:
+            self._log_deletion(conversation_id, True)
+
+    def save_lock(self, lock: TicketLock) -> None:
+        """Commit individual tickets and last issued ticket number to storage."""
+        last_issued_ticket = lock.tickets[-1]
+        serialised_ticket = last_issued_ticket.dumps()
+        key = (
+            self.key_prefix
+            + lock.conversation_id
+            + ":"
+            + str(last_issued_ticket.number)
+        )
+        self.red.set(
+            name=key, value=serialised_ticket, ex=int(last_issued_ticket.expires)
+        )
+
+        # set expiry on the last_issued_ticket_number key too
+        last_issued_key = (
+            self.key_prefix
+            + lock.conversation_id
+            + ":"
+            + LAST_ISSUED_TICKET_NUMBER_SUFFIX
+        )
+        self.red.expireat(name=last_issued_key, when=int(last_issued_ticket.expires))
+
+    def increment_ticket_number(self, lock: TicketLock) -> int:
+        """Uses Redis atomic transaction to increment ticket number."""
+        last_issued_key = (
+            self.key_prefix
+            + lock.conversation_id
+            + ":"
+            + LAST_ISSUED_TICKET_NUMBER_SUFFIX
+        )
+
+        # if last_issued_ticket_number value is not set or has expired
+        # we need to (re)set the key and then increment it
+
+        last_issued_ticket_number = self.red.get(last_issued_key)
+
+        if last_issued_ticket_number is None:
+            self.red.set(name=last_issued_key, value=lock.last_issued)
+
+        return self.red.incr(name=last_issued_key)
```

Comment:
incr should default to setting the key to 0 if it doesn't already exist. I don't think the get and if before this statement is needed or is there a particular reason why you added this?

Reply:
As per [my comment](https://github.com/RasaHQ/rasa/pull/11256#discussion_r916695614) here, I don't think `redis-py` retained this behaviour described in the redis docs. I can try this suggestion again as a sanity check 😄

Reply:
After trying this again, it worked 😄 Not sure what I was doing before, but comment addressed: https://github.com/RasaHQ/rasa/pull/11256/commits/98eadadbb7a0144960b8702b67008627e7fe6816

Reply:
🔥